syntax = "proto3";
option cc_generic_services = true;
package pb;

//消息枚举
enum MessageID
{
    MSG_NONE = 0;
    MSG_SYNC_PID = 1;
    MSG_TALK = 2;
    MSG_POSITION = 3;
    MSG_BROADCAST = 200;
    MSG_SYNCPID = 201;
    MSG_SYNC_PLAYERS = 202;
    MSG_LOGIN_REQUEST = 1001;
    MSG_LOGIN_REPLY = 1002;
    
}
message SynxPid
{
    int32 Pid = 1;
}
message Talk
{
    string Content = 1;
}
message Position
{
    float X = 1;
    float Y = 2;
    float Z = 3;
    float V = 4;
}
message BroadCast
{
    //Tp: 1 世界聊天, 2 坐标, 3 动作, 4 移动之后坐标信息更新
    int32 Pid=1;
    int32 Tp=2;
    oneof Data {
        string Content=3;
        Position P=4;
        int32 ActionData=5;
    }
}
message SyncPid
{
    int32 Pid=1;
}
message SyncPlayers
{
    //嵌套多个子消息类型Player的消息
    repeated Player ps=1;
}
message Player
{
    int32 Pid=1;
    Position P=2;
}
